packages/wallet-sdk/src/sign/scw/utils.ts

```diff
@@ -138,3 +142,80 @@ export async function initSubAccountConfig() {
     capabilities,
   });
 }
+
+export function assertFetchPermissionsRequest(
+  request: RequestArguments
+): asserts request is FetchPermissionsRequest | EmptyFetchPermissionsRequest {
+  if (request.method === 'coinbase_fetchPermissions' && request.params === undefined) {
+    return;
+  }
+
+  if (
+    request.method === 'coinbase_fetchPermissions' &&
+    Array.isArray(request.params) &&
+    request.params.length === 1 &&
+    typeof request.params[0] === 'object'
+  ) {
+    if (
+      typeof request.params[0].account !== 'string' ||
+      !request.params[0].chainId.startsWith('0x')
+    ) {
+      throw standardErrors.rpc.invalidParams(
+        'FetchPermissions - Invalid params: params[0].account must be a hex string'
+      );
+    }
+
+    if (
+      typeof request.params[0].chainId !== 'string' ||
+      !request.params[0].chainId.startsWith('0x')
+    ) {
+      throw standardErrors.rpc.invalidParams(
+        'FetchPermissions - Invalid params: params[0].chainId must be a hex string'
+      );
+    }
+
+    if (
+      typeof request.params[0].spender !== 'string' ||
+      !request.params[0].spender.startsWith('0x')
+    ) {
+      throw standardErrors.rpc.invalidParams(
+        'FetchPermissions - Invalid params: params[0].spender must be a hex string'
+      );
+    }
+
+    return;
+  }
+
+  throw standardErrors.rpc.invalidParams();
+}
+
+export function fillMissingParamsForFetchPermissions(
+  request: FetchPermissionsRequest | EmptyFetchPermissionsRequest
+): FetchPermissionsRequest {
+  if (request.params !== undefined) {
+    return request as FetchPermissionsRequest;
+  }
+
+  // this is based on the assumption that the first account is the active account
+  // it could change in the context of multi-(universal)-account
+  const accountFromStore = store.getState().account.accounts?.[0];
+  const chainId = store.getState().account.chain?.id;
+  const subAccountFromStore = store.getState().subAccount?.address;
+
+  if (!accountFromStore || !subAccountFromStore || !chainId) {
+    throw standardErrors.rpc.invalidParams(
+      'FetchPermissions - one or more of account, sub account, or chain id is missing, connect to sub account via wallet_connect first'
+    );
+  }
+
+  return {
+    method: 'coinbase_fetchPermissions',
+    params: [
+      {
+        account: accountFromStore,
+        chainId: numberToHex(chainId),
```

Comment:
might be a good idea to check if the chainId is in fact hex before doing this. wrapper libs like viem tend format numbers in RPC as hex before passing it along